python/ray/_raylet.pyx

```diff
@@ -332,16 +332,17 @@ class StreamingObjectRefGenerator:
             timeout_s: If the next object is not ready within
                 this timeout, it returns the nil object ref.
         """
-        self.worker.check_connected()
         core_worker = self.worker.core_worker
 
         # Wait for the next ObjectRef to become ready.
-        expected_ref = core_worker.peek_object_ref_stream(
+        expected_ref, is_ready = core_worker.peek_object_ref_stream(
             self._generator_ref)
-        ready, unready = ray.wait(
```

Comment:
Could you explain why `ray.wait()` ready object is expensive even when `fetch_local` is false?

Reply:
You mean add comments from code?

It is because the router itself has very high load (10s of thousand of iteration/s). ray.wait itself only has 200~500us overhead, but it is expensive when you have 10s of thousand of tasks. If we skip this, we can do this process within 10~20 us